### PR TITLE
owner(ticdc): fix prometheus panic

### LINKED ISSUE #4742: TiCDC owner panics on the master branch
```
panic: inconsistent label cardinality: expected 2 label values but got 3 in []string{"test-1", "new-sched-test-cdc-6.liuzixiong.tispace:8300", "total"}

goroutine 139 [running]:
github.com/prometheus/client_golang/prometheus.(*GaugeVec).WithLabelValues(0xc000a08100, 0xc04cfac600, 0x3, 0x3, 0xc001e380a0, 0xc04cfa8b01)
	github.com/prometheus/client_golang@v1.7.1/prometheus/gauge.go:217 +0xff
github.com/pingcap/tiflow/cdc/owner.(*ownerImpl).updateMetrics(0xc000d3b420, 0xc002f5af40)
	github.com/pingcap/tiflow/cdc/owner/owner.go:355 +0x930
github.com/pingcap/tiflow/cdc/owner.(*ownerImpl).Tick(0xc000d3b420, 0x7f4e39d7b600, 0xc01a377a20, 0x3f0db88, 0xc002f5af40, 0x1, 0x0, 0x1, 0x0)
	github.com/pingcap/tiflow/cdc/owner/owner.go:155 +0x87
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc002c7f480, 0x7f4e39d7b600, 0xc01a377a20, 0xc000f7df50, 0xbebc200, 0x38335da, 0x5, 0x0, 0x0)
	github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:237 +0x691
github.com/pingcap/tiflow/cdc/capture.(*Capture).runEtcdWorker(0xc00056c7e0, 0x3f5d508, 0xc01a377a20, 0x7f4e38bff390, 0xc000d3b420, 0x3f0db88, 0xc002f5af40, 0xbebc200, 0x38335da, 0x5, ...)
	github.com/pingcap/tiflow/cdc/capture/capture.go:464 +0x14f
github.com/pingcap/tiflow/cdc/capture.(*Capture).campaignOwner(0xc00056c7e0, 0x3f5d508, 0xc0007fe100, 0x3, 0xc000965e00)
	github.com/pingcap/tiflow/cdc/capture/capture.go:437 +0xa6f
github.com/pingcap/tiflow/cdc/capture.(*Capture).run.func2(0xc000736060, 0xc00056c7e0, 0x3f5d508, 0xc0007fe100, 0xc000f1e000)
	github.com/pingcap/tiflow/cdc/capture/capture.go:301 +0xb5
created by github.com/pingcap/tiflow/cdc/capture.(*Capture).run
	github.com/pingcap/tiflow/cdc/capture/capture.go:295 +0x37b

```

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #4742

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test

Run TiCDC withe new scheduler enabled.

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

**Comment by 3AceShowHand** [comment]
please check the code twice to make sure no more related misbehavior in the codebase, then we can merge it.

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -351,11 +351,11 @@ func (o *ownerImpl) updateMetrics(state *orchestrator.GlobalReactorState) {
 			totalCounts := infoProvider.GetTotalTableCounts()
 			pendingCounts := infoProvider.GetPendingTableCounts()
 
-			for captureID, info := range o.captures {
+			for captureID := range o.captures {
 				ownerMaintainTableNumGauge.WithLabelValues(
-					cfID, info.AdvertiseAddr, maintainTableTypeTotal).Set(float64(totalCounts[captureID]))
+					cfID, maintainTableTypeTotal).Set(float64(totalCounts[captureID]))
```
Comment (liuzix) [inline cdc/owner/owner.go:356]: We need the `captureAddr` label here.

**Review by liuzix — CHANGES_REQUESTED** [review]

**Review by liuzix — APPROVED** [review]

**Pull request merged**